packages/mermaid/src/diagrams/ishikawa/ishikawaDb.ts

```diff
@@ -0,0 +1,199 @@
+import { getConfig } from '../../diagram-api/diagramAPI.js';
+import type { D3Element } from '../../types.js';
+import { sanitizeText } from '../../diagrams/common/common.js';
+import { log } from '../../logger.js';
+import type { IshikawaNode } from './ishikawaTypes.js';
+import defaultConfig from '../../defaultConfig.js';
+
+let nodes: IshikawaNode[] = [];
+let cnt = 0;
+let elements: Record<number, D3Element> = {};
+let problemStatement = '';
+let categories: string[] = [];
```

Comment:
We need to use a class based approach for DB.

ref: 

- https://github.com/mermaid-js/mermaid/pull/6804
- https://github.com/mermaid-js/mermaid/pull/6803